content/blog/migrations-with-atlas-and-cloudnativepg/index.md

```diff
@@ -0,0 +1,203 @@
+from the AtlasSchema CRD to reference
+the password and the service. Following along the Atlas Operator Quickstart, we
+create a migration defining a table called `t1`. Save the following to a file
+named `atlas-schema.yaml`.
+Notice how CloudNativePG automatically produced the values we need to put
+in the `passwordFrom` and `host` fields:
+
+``` yaml
+apiVersion: db.atlasgo.io/v1alpha1
+kind: AtlasSchema
+metadata:
+  name: atlasschema-pg
+spec:
+  credentials:
+    scheme: postgres
+    host: cluster-example-rw.default
+    user: app
+    passwordFrom:
+      secretKeyRef:
+        key: password
+        name: cluster-example-app
+    database: app
+    port: 5432
+    parameters:
+      sslmode: disable
+  schema:
+    sql: |
+      create table t1 (
+        id int
+      );
+```
+
+Then apply it: `kubectl apply -f atlas-schema.yaml`.
+
+In a few seconds, you should be able to see the Schema has been reconciled:
+
+``` console
+> kubectl get atlasschemas.db.atlasgo.io
+NAME                 READY   REASON
+atlasschema-pg       True    Applied
+```
+
+To verify the effects on the database, let's get a `psql` session open on one
+of our instances:
+
+``` console
+> kubectl exec -ti cluster-example-1 -- psql app
+Defaulted container "postgres" out of: postgres, bootstrap-controller (init)
+psql (17.4 (Debian 17.4-1.pgdg110+2))
+Type "help" for help.
+
+app=# \dt
+       List of relations
+ Schema | Name | Type  | Owner
+--------+------+-------+-------
+ public | t1   | table | app
+(1 row)
+
+app=# \d t1
+                 Table "public.t1"
+ Column |  Type   | Collation | Nullable | Default
+--------+---------+-----------+----------+---------
+ id     | integer |           |          |
+```
+
+As suggested in the Atlas Quickstart, we can modify the schema in the
+`atlas-schema.yaml` file, and then re-apply with
+`kubectl apply -f atlas-schema.yaml`, and the Atlas operator will again
+reconcile the database to the desired state.
+
+## What's the point? DevOps
+
+Let's say you have an application that uses a database. In production, the
+database may hold your customers's data, purchase history, financial
+transactions, perhaps maps using PostGIS, etc.
+
+Application developers will regularly need to add functionality, and often
+that will involve creating new tables, schemas, procedures, indexes, perhaps
+doing some `INSERT` statements to populate static data, etc.
+The developers will be using a development database, probably hosted in their
+dev machines.
+There might be another database for the purpose of automated testing, and
+of course there's the production database, where the changes will need to be
+applied in the end. These different databases will have different data in them,
+and different loads.
+Add time and other developers, and you have a big chance for snafus of the type
+*"but it worked on my machine!"*
+
+Database migration tools manage this, bringing DevOps to this area of databases.
```

Comment:
I would link directly this capability from DORA: https://dora.dev/capabilities/database-change-management/

Reply:
Thanks, that's a great idea. I added a paragraph.